### PR TITLE
Rebuild with OpenSSL 3

### PR DESCRIPTION
**Pull request by JeanChristopheMorinPerso**
Rebuild with OpenSSL 3. Also did some cleanup of the dependencies.

Depends on:
* https://github.com/AnacondaRecipes/postgresql-feedstock/pull/6

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -24,10 +24,13 @@ requirements:
...
-    - openssl
-    - postgresql
     - wheel
-    - krb5  # [win]
+    # It's technically not required by the code, but the do excplitly link with -lcrypto.
+    # The linker is smart enough to not actually link it since no openssl symbol is directly used,
+    # hence why we also ad it to ignore_run_exports.
+    - openssl {{ openssl }}  # [not linux]
+    - libpq 12.15
+    - krb5 1.20.1  # [win]
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: This is wrong... We should rebuild postgres (and libpq) so thta it has a dependency on krb5.

The libpq build says:
```
INFO (libpq,Library/bin/libpq.dll): Needed DSO $RPATH/gssapi64.dll found in the whitelist
```
And when we look at the [recipe](https://github.com/AnacondaRecipes/postgresql-feedstock/blob/master/recipe/meta.yaml#L79-L80), we can see that they are indeed ignored. That's bad.

Additionally, I see
```
INFO  WARNING (libpq,Library/bin/libpq.dll): $RPATH/libssl-3-x64.dll not found in packages, sysroot(s) nor the missing_dso_whitelist.                                                                                                                 
INFO  .. is this binary repackaging?                                                                                                                                                                                                                  
INFO  WARNING (libpq,Library/bin/libpq.dll): $RPATH/libcrypto-3-x64.dll not found in packages, sysroot(s) nor the missing_dso_whitelist.                                                                                                              
INFO  .. is this binary repackaging?           
```

**Review by skupr-anaconda — APPROVED** [review]
LGTM

File: recipe/meta.yaml [hunk 2]
```diff
@@ -24,14 +21,12 @@ requirements:
...
     - pip
-    - openssl
-    - postgresql
     - wheel
-    - krb5  # [win]
+    - setuptools
+    - libpq 12.15
   run:
     - python
     - libpq
```
Comment (chenghlee) [inline recipe/meta.yaml:29]: Does `libpq` have a run export for the version? If not, we should probably update this to be `libpq >=X.Y` to match the `host` requirement so users don't accidentally end up using a `libpq` version older than the one we build with.

Comment (chenghlee) [reply recipe/meta.yaml:29]: Never mind. I should have just looked at the PostgreSQL recipe to answer it for myself. 😄

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:29]: It does yes It exports `>=12,<13`.

**Review by cbouss — APPROVED** [review]
LGTM

**Pull request merged**